### PR TITLE
Flaps in prelaunch

### PR DESCRIPTION
**Pull request by smr277**
Fixes an issue where Takeoff Flaps aren't set in prelaunch (if a plane waits for an acceleration or speed to go into the takeoff flight_stage).

### CODE REVIEW

File: ArduPlane/Parameters.cpp [hunk 1]
```diff
@@ -225,6 +225,7 @@ const AP_Param::Info Plane::var_info[] = {
     // @User: Advanced
     GSCALAR(takeoff_flap_percent,     "TKOFF_FLAP_PCNT", 0),
 
+
```
Comment (magicrub) [inline ArduPlane/Parameters.cpp]: please remove this whitespace

File: ArduPlane/Attitude.cpp [hunk 2]
```diff
@@ -1054,6 +1054,9 @@ void Plane::set_servos(void)
                 }
                 break;
             default:
+                if (in_preLaunch_flight_stage()) {
```
Comment (magicrub) [inline ArduPlane/Attitude.cpp]: default includes quadplane stuff, please make this _NORMAL only

Comment (magicrub) [inline ArduPlane/Attitude.cpp]: it also needs to check g.takeoff_flap_percent != 0 like above

File: ArduPlane/Parameters.h [hunk 3]
```diff
@@ -124,7 +124,6 @@ class Parameters {
         k_param_rangefinder_landing,
         k_param_land_flap_percent,
         k_param_takeoff_flap_percent,
-        k_param_takeoff_flap_prelaunch,
```
Comment (magicrub) [inline ArduPlane/Parameters.h]: btw, the problem with your eeprom was you added this flap_prelaunch right in the middle of this enum and pushed all the other indexes down which corrupted the memory indexes.

**Comment by WickedShell** [comment]
Would it make more sense to add another flight stage? IE TAKEOFF_WAIT? I'm a bit worried to see the default case handle that there, and it would also allow for a easier approach of mixing in any other delays based on waiting for the takeoff.

**Comment by magicrub** [comment]
@WickedShell yes, and I have an old branch of that but we need to re-write it. I had Event38 add it when they were working on reverse thrust late last year.

**Comment by smr277** [comment]
Also - just want to say we've done 10+ flights and 5+ launches with this code. Works beautifully. 

I would also suggest adding a TAKEOFF_WAIT mode. Are people interested in this?

**Comment by magicrub** [comment]
yes, interested

**Comment by smr277** [comment]
Is there anything else I need to do to get this merged in?

**Comment by OXINARF** [comment]
@smr277 This isn't passing tests. Also you shouldn't have merge commits. You need to do an interactive rebase on master. If you need help doing that don't hesitate to use Gitter.

**Comment by magicrub** [comment]
fixed and merged and tweaked to be https://github.com/ArduPilot/ardupilot/commit/ee29f38cdfd7f6c717651846ff23b3304bd83e25

**Pull request closed without merging**